api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -452,6 +454,68 @@ private static void checkSchemaCompatibility(
     }
   }
 
+  /**
+   * Estimates the number of bytes a value for a given field may occupy in memory.
+   *
+   * <p>This method approximates the memory size based on the internal Java representation defined
+   * by {@link Type.TypeID}. It is important to note that the actual size might differ from this
+   * estimation. The method is designed to handle a variety of data types, including primitive
+   * types, strings, and nested types such as structs, maps, and lists.
+   *
+   * @param field a field for which to estimate the size
+   * @return the estimated size in bytes of the field's value in memory
+   */
+  public static long defaultSize(Types.NestedField field) {
+    return defaultSize(field.type());
+  }
+
+  private static long defaultSize(Type type) {
+    switch (type.typeId()) {
+      case BOOLEAN:
+        // the size of a boolean variable is virtual machine dependent
+        // it is common to believe booleans occupy 1 byte in most JVMs
+        return 1;
+      case INTEGER:
+      case FLOAT:
+      case DATE:
+        // ints and floats occupy 4 bytes
+        // dates are internally represented as ints
+        return 4;
+      case LONG:
+      case DOUBLE:
+      case TIME:
+      case TIMESTAMP:
+        // longs and doubles occupy 8 bytes
+        // times and timestamps are internally represented as longs
+        return 8;
+      case STRING:
+        // 12 (header) + 12 (fields) + 16 (array overhead) + 20 (10 chars, 2 bytes each) = 60 bytes
+        return 60;
+      case UUID:
+        // 12 (header) + 16 (two long variables) = 28 bytes
+        return 28;
```

Comment:
should we express these as OBJECT_HEADER

Reply:
Do you mean like an actual expression?

Reply:
I mean like OBJECT_HEADER + 16?

Reply:
Comment is still not addressed?

Reply:
After thinking a bit more, I am not sure it would make it any better. The second number would still be some abstract number. I feel it should be OK to just explain how the number was derived and use the constant for nested types. We have real computations there.